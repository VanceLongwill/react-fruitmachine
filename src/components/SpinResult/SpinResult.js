import React from 'react'
import styled from 'styled-components'

const ResultsContainer = styled.div`
  height: 100px;
  display: flex;
  flex-direction: column;
`

const StyledText = styled.p`
  color: white;
  font-size: 24px;
  font-weight: 700;
  width: 100%;
  padding: 20px;
  background-color: ${p => p.isWin ? 'green' : 'red'}
`

function getResultText(count, color) {
  switch(count) {
    case 3:
      return `Congratulations! You won the jackpot with ${count} ${color} squares`
    case 2:
      return `Unlucky! You only matched ${count} ${color} squares`
    default:
      return `Unlucky! You didn't get any matches`
  }
}

export function SpinResult(props) {
  const { results } = props;
  const notifications = results.map(result => (
    <StyledText isWin={result.count === 3}>
      {getResultText(result.count, result.color)}
    </StyledText>
  ))
  return (
    <ResultsContainer>
      {notifications}
    </ResultsContainer>
  )
}
