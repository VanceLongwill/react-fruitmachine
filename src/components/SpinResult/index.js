import { SpinResult } from './SpinResult'
export default SpinResult
