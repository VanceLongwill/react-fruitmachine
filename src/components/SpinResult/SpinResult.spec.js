import React from 'react'
import { mount } from 'enzyme'

import { SpinResult } from './SpinResult'

describe('SpinResult', () => {
  const tests = [
    [{ key: 'some-uuid', color: 'blue', count: 3 }],
    [{ key: 'some-uuid', color: 'blue', count: 2 }],
    [{ key: 'some-uuid', color: 'blue', count: 1 }, { key: 'some-uuid', color: 'blue', count: 2 }],
  ]
  tests.forEach((test, i) => {
    it(`should match snapshot - ${i}`, () => {
      expect(mount(<SpinResult results={test}/>)).toMatchSnapshot()
    })
  })
})
