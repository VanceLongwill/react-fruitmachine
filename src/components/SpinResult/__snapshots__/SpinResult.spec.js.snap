// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinResult should match snapshot - 0 1`] = `
<styled.div>
  <div
    className="sc-bdVaJa bzgQpZ"
  >
    <styled.p
      count={3}
    >
      <p
        className="sc-bwzfXH frQxat"
      >
        Jackpot! You won with 3 blue squares
      </p>
    </styled.p>
  </div>
</styled.div>
`;

exports[`SpinResult should match snapshot - 1 1`] = `
<styled.div>
  <div
    className="sc-bdVaJa bzgQpZ"
  >
    <styled.p
      count={2}
    >
      <p
        className="sc-bwzfXH cBSDyN"
      >
        Almost! You only matched 2 blue squares
      </p>
    </styled.p>
  </div>
</styled.div>
`;

exports[`SpinResult should match snapshot - 2 1`] = `
<styled.div>
  <div
    className="sc-bdVaJa bzgQpZ"
  >
    <styled.p
      count={2}
    >
      <p
        className="sc-bwzfXH cBSDyN"
      >
        Almost! You only matched 2 blue squares
      </p>
    </styled.p>
  </div>
</styled.div>
`;
